Crash when resizing the app in tablets

**Reproduction steps**:
1. Open the app in a device large enough to make the deck picker "fragmented"
2. Use Android's split screen feature to open another app on the side

**Expected behavior**

Anki shouldn't crash


**Trace:**

```text
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki.debug/com.ichi2.anki.DeckPicker}: java.lang.NullPointerException
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4454)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4685)
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:7028)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:6906)
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:79)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:60)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:174)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:109)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:81)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2895)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:257)
	at android.os.Looper.loop(Looper.java:342)
	at android.app.ActivityThread.main(ActivityThread.java:9634)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:619)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:929)
Caused by: java.lang.NullPointerException
	at com.ichi2.anki.DeckPicker.onCreate(DeckPicker.kt:625)
               	at android.app.Activity.performCreate(Activity.java:9363)
```

**Debug info**
```text
AnkiDroid Version = 2.23.0alpha2-debug (1c192d685c09a9c520aabda6c4dc7ebaa1b57e5a)  
Backend Version = 0.1.60-anki25.07.5 (25.07.5 7172b2d26684c7ef9d10e249bd43dc5bf73ae00c)  
Android Version = 15 (SDK 35)  
ProductFlavor = full  
Device Info = samsung | samsung | gts9 | gts9xxx | SM-X716B | qcom  
Webview User Agent = Mozilla/5.0 (Linux; Android 15; SM-X716B Build/AP3A.240905.015.A2; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/139.0.7258.94 Safari/537.36  
ACRA UUID = 8f702b06-70e0-44e5-bb86-bdd8c70eea80  
FSRS = 4.1.1 (Enabled: true)  
Crash Reports Enabled = false
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
@@ -71,8 +71,6 @@ import androidx.core.view.doOnLayout
 import androidx.core.view.isVisible
 import androidx.draganddrop.DropHelper
 import androidx.fragment.app.commit
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.flowWithLifecycle
 import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.LinearLayoutManager
@@ -251,23 +249,12 @@ open class DeckPicker :
     ApkgImportResultLauncherProvider,
     CsvImportResultLauncherProvider,
     CollectionPermissionScreenLauncher {
-    val viewModel: DeckPickerViewModel by viewModels {
-        object : ViewModelProvider.Factory {
-            override fun <T : ViewModel> create(modelClass: Class<T>): T {
-                val fragmented =
-                    (
-                        resources.configuration.screenLayout and
-                            Configuration.SCREENLAYOUT_SIZE_MASK
-                    ) ==
-                        Configuration.SCREENLAYOUT_SIZE_XLARGE
-                @Suppress("UNCHECKED_CAST")
-                return DeckPickerViewModel(fragmented) as T
-            }
-        }
-    }
+    val viewModel: DeckPickerViewModel by viewModels()
 
     override var fragmented: Boolean
-        get() = viewModel.fragmented
+        get() =
+            resources.configuration.screenLayout and Configuration.SCREENLAYOUT_SIZE_MASK ==
+                Configuration.SCREENLAYOUT_SIZE_XLARGE
         set(_) = throw UnsupportedOperationException()
 
     // Short animation duration from system
@@ -800,9 +787,9 @@ open class DeckPicker :
             }
         }
 
-        fun onStudyOptionsVisibilityChanged(isVisible: Boolean) {
+        fun onStudyOptionsVisibilityChanged(collectionHasNoCards: Boolean) {
             invalidateOptionsMenu()
-            studyoptionsFrame?.isVisible = isVisible
+            studyoptionsFrame?.isVisible = fragmented && !collectionHasNoCards
         }
 
         fun onDeckListChanged(deckList: FlattenedDeckList) {
@@ -876,7 +863,7 @@ open class DeckPicker :
         viewModel.flowOfStudiedTodayStats.launchCollectionInLifecycleScope(::onStudiedTodayChanged)
         viewModel.flowOfDeckListInInitialState.filterNotNull().launchCollectionInLifecycleScope(::onCollectionStatusChanged)
         viewModel.flowOfCardsDue.launchCollectionInLifecycleScope(::onCardsDueChanged)
-        viewModel.flowOfStudyOptionsVisible.launchCollectionInLifecycleScope(::onStudyOptionsVisibilityChanged)
+        viewModel.flowOfCollectionHasNoCards.launchCollectionInLifecycleScope(::onStudyOptionsVisibilityChanged)
         viewModel.flowOfDeckList.launchCollectionInLifecycleScope(::onDeckListChanged)
         viewModel.flowOfFocusedDeck.launchCollectionInLifecycleScope(::onFocusedDeckChanged)
         viewModel.flowOfResizingDividerVisible.launchCollectionInLifecycleScope(::onResizingDividerVisibilityChanged)

--- a/AnkiDroid/src/main/java/com/ichi2/anki/deckpicker/DeckPickerViewModel.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/deckpicker/DeckPickerViewModel.kt
@@ -51,7 +51,6 @@ import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
@@ -60,12 +59,9 @@ import timber.log.Timber
 
 /**
  * ViewModel for the [DeckPicker]
- *
- * @param fragmented whether the study options is shown alongside the deck picker
  */
-class DeckPickerViewModel(
-    val fragmented: Boolean,
-) : ViewModel(),
+class DeckPickerViewModel :
+    ViewModel(),
     OnErrorListener {
     val flowOfStartupResponse = MutableStateFlow<StartupResponse?>(null)
 
@@ -163,8 +159,6 @@ class DeckPickerViewModel(
 
     /** "Studied N cards in 0 seconds today */
     val flowOfStudiedTodayStats = MutableStateFlow("")
-
-    val flowOfStudyOptionsVisible = flowOfCollectionHasNoCards.map { noCards -> fragmented && !noCards }
 
     /** Flow that determines when the resizing divider should be visible */
     val flowOfResizingDividerVisible =

--- a/AnkiDroid/src/test/java/com/ichi2/anki/deckpicker/DeckPickerViewModelTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/deckpicker/DeckPickerViewModelTest.kt
@@ -39,7 +39,7 @@ import timber.log.Timber
 /** Test of [DeckPickerViewModel] */
 @RunWith(AndroidJUnit4::class)
 class DeckPickerViewModelTest : RobolectricTest() {
-    private val viewModel = DeckPickerViewModel(fragmented = false)
+    private val viewModel = DeckPickerViewModel()
 
     @Test
     fun `empty cards - flow`() =

